code/modules/detectivework/tools/evidencebag.dm

```diff
@@ -9,90 +7,89 @@
 	w_class = ITEM_SIZE_SMALL
 	var/obj/item/stored_item = null
 
-/obj/item/weapon/evidencebag/MouseDrop(obj/item/I as obj)
-	if (!ishuman(usr))
-		return
-
+/obj/item/weapon/evidencebag/MouseDrop(obj/item/I)
 	var/mob/living/carbon/human/user = usr
 
-	if (!(user.l_hand == src || user.r_hand == src))
-		return //bag must be in your hands to use
+	if (!ishuman(user))
+		return
 
 	if (isturf(I.loc))
 		if (!user.Adjacent(I))
 			return
-	else
-		//If it isn't on the floor. Do some checks to see if it's in our hands or a box. Otherwise give up.
-		if(istype(I.loc,/obj/item/weapon/storage))	//in a container.
-			var/sdepth = I.storage_depth(user)
-			if (sdepth == -1 || sdepth > 1)
-				return	//too deeply nested to access
-
-			var/obj/item/weapon/storage/U = I.loc
-			user.client.screen -= I
-			U.contents.Remove(I)
-		else if(user.l_hand == I)					//in a hand
-			user.drop_l_hand()
-		else if(user.r_hand == I)					//in a hand
-			user.drop_r_hand()
-		else
-			return
 
 	if(!istype(I) || I.anchored)
 		return
 
+	if(istype(I.loc,/obj/item/weapon/storage))	//in a container.
+		var/sdepth = I.storage_depth(user)
+		if (sdepth == -1 || sdepth > 1)
+			return	//too deeply nested to access
+
+		var/obj/item/weapon/storage/U = I.loc
+		user.client.screen -= I
+		U.contents.Remove(I)
+	else if(user.l_hand == I && (get_dist(src, I) == 0))					//in a hand
+		user.drop_l_hand()
+		qdel(I)
+	else if(user.r_hand == I && (get_dist(src, I) == 0))					//in a hand
+		user.drop_r_hand()
+		qdel(I)
+	else
+		if(get_dist(src, I) > 1)
+			return
+
 	if(istype(I, /obj/item/weapon/evidencebag))
-		to_chat(user, "<span class='notice'>You find putting an evidence bag in another evidence bag to be slightly absurd.</span>")
+		to_chat(user, SPAN_NOTICE("You find putting an evidence bag in another evidence bag to be slightly absurd."))
```

Comment:
SPAN("notice", "You find putting an evidence bag in another evidence bag to be slightly absurd.")